lib/master.js

```diff
@@ -143,6 +143,23 @@ function Master(options, adapter) {
         }
     }
 
+    function pollBinariesBlock_keep() {
+        if (options.config.keepAliveInterval > 0) {
+            if (modbusClient) {
+                modbusClient['readDiscreteInputs'](deviceIds[0], 0, 1)
+                    .then(response => {
+                        adapter.log.silly('Keep alive response = ' + JSON.stringify(response));
+                    })
+                    .catch(() => {});
+            } else {
+                adapter.log.debug(`Poll canceled, because no connection`);
+            }
+            setTimeout(pollBinariesBlock_keep, options.config.keepAliveInterval);
```

Comment:
and also use this variable to detect if the timeout is already set because the code that calls this function has no protection against potentially building up multiple timers just because connection is "flapping".